packages/report-server/src/reportBuilder/transform/functions/aggregate/aggregations.ts

```diff
@@ -3,75 +3,124 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import { Row, FieldValue } from '../../../types';
+import { FieldValue } from '../../../types';
 
 const isUndefined = (value: FieldValue): value is undefined => {
   return value === undefined;
 };
 
-const group = (existingRow: Row, field: string, value: FieldValue) => {
-  existingRow[field] = value;
+const filterUndefinedAndNull = (values: FieldValue[]): NonNullable<FieldValue[]> => {
+  const filteredValues: NonNullable<FieldValue[]> = [];
+  values.forEach(value => {
+    if (value !== undefined && value !== null) {
+      filteredValues.push(value);
+    }
+  });
+  return filteredValues;
```

Comment:
With a function like this, I'd argue it's better to use the standard `Array.filter` function. Something like this:
```
const isNotNull = <T>(value: T): value is NotNullable<T> => value !== undefined && value !== null;
``` 
^here we are using a generic function to accept any type, and return a boolean to say whether the type is not null

and then later it can be used like:
```
const notNullValues = values.filter(isNotNull);
```